src/open_samus_returns_rando/lua_editor.py

```diff
@@ -284,6 +285,19 @@ def save_modifications(self, editor: PatcherEditor, configuration: dict):
             []
         )
 
+        editor.add_new_asset(
+            "system/scripts/cosmetics.lc",
+            Lua(Container(lua_text=files_path().joinpath("templates", "cosmetics.lua").read_text()),
+                editor.target_game),
+            []
+        )
+
+        cosmetics_script = cosmetic_patches._lua_cosmetics(configuration["cosmetic_patches"])
+        editor.replace_asset(
+            "system/scripts/cosmetics.lc",
+            Lua(Container(lua_text=cosmetics_script), editor.target_game)
+        )
```

Comment:
Simplify the whole block here.
It's adding a new asset with the `TEMPLATE` variables in the lua, then gets the whole script with the replaced template and then replaces the asset with the content from the step before.

You can directly add the asset with the correct content.